.github/workflows/ci_build_android.yml

```
@@ -0,0 +1,50 @@
+name: Build Android library
+
+on:
+  workflow_call:
+    outputs:
+      artifact-dir-base:
+        description: The base name of artifact directory (without the target suffix)
+        value: ${{ jobs.build-android-libs.outputs.artifact-dir-base }}
+
+jobs:
+  build-android-libs:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 35
+    strategy:
+      matrix:
+        target: ["arm64-v8a", "x86_64"]
+    env:
+      ARTIFACT_DIR_BASE: ${{ github.job }}
+    outputs:
+      artifact-dir-base: ${{ env.ARTIFACT_DIR_BASE }}
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Setup the rust CI.
+        uses: ./.github/actions/setup-job
+        with:
+          rust: true
+          dart: true
+          android: true
+          rust-target: "aarch64-linux-android x86_64-linux-android"
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build Android library
+        run: |
+          export PATH="$(pwd)/cargo-installs/bin:${PATH}"
+          just compile-android "${{ matrix.target }}"
+
+      - name: Prepare lib for upload
+        id: lib
+        run: |
+          FILE=${{ env.FLUTTER_WORKSPACE }}/android/src/main/jniLibs/${{ matrix.target }}/libxayn_discovery_engine_bindings.so
```

we can also use 
```suggestion
          FILE=${{ env.FLUTTER_WORKSPACE }}/android/src/main/jniLibs/${{ matrix.target }}/${{ env.IOS_LIB_BASE }}.so
```
but then we should better remame it to `LIB_BASE` or something like that

---

Outdated now because we'll upload the entire folder.